### ISSUE TITLE
Fails to build on windows

### ISSUE THREAD

**Issue by cpyarger** [obsproject/obs-plugintemplate#35]
### Operating System Info

Windows 11

### OBS Studio Version

Git

### OBS Studio Log URL

 n/a

### Expected Behavior

Successfully builds plugin template locally

### Current Behavior

```
PS C:\Users\cpyar\Desktop\obs\obs-plugintemplate>  .\.github\scripts\Build-Windows.ps1
     >   Found dependency 7z as C:\Program Files\7-zip\7z.exe
     >   Found dependency cmake as C:\Program Files\CMake\bin\cmake.exe
     >   Found dependency iscc as C:\Program Files (x86)\Inno Setup 6\ISCC.exe
    =>   Setting up Pre-built obs-deps...
     >   Found downloaded Pre-built obs-deps.
     >   Checksum of downloaded Pre-built obs-deps matches.
    =>   Setting up Pre-built Qt6...
     >   Found downloaded Pre-built Qt6.
     >   Checksum of downloaded Pre-built Qt6 matches.
    =>   Checking for Git executable...
     >   Git found.
    =>   Setting up OBS Studio...
HEAD is now at 67f7a8407 CI: Change build file names
    =>   Checked out commit 67f7a84076be2c73d172eb14265c60d6209598db on branch master
Entering 'plugins/enc-amf'
Entering 'plugins/mac-syphon/syphon-framework'
Entering 'plugins/obs-browser'
Entering 'plugins/obs-outputs/ftl-sdk'
Synchronizing submodule url for 'libcurl'
Synchronizing submodule url for 'libjansson'
Entering 'plugins/obs-outputs/ftl-sdk/libcurl'
Entering 'plugins/obs-outputs/ftl-sdk/libjansson'
Entering 'plugins/obs-websocket'
Synchronizing submodule url for 'deps/asio'
Synchronizing submodule url for 'deps/json'
Synchronizing submodule url for 'deps/qr'
Synchronizing submodule url for 'deps/websocketpp'
Entering 'plugins/obs-websocket/deps/asio'
Entering 'plugins/obs-websocket/deps/json'
Entering 'plugins/obs-websocket/deps/qr'
Entering 'plugins/obs-websocket/deps/websocketpp'
Entering 'plugins/win-dshow/libdshowcapture'
    =>   Configuring OBS Studio...
    =>   Configuring OBS...
CMake Error: No generator specified for -G

Generators
* Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 12 2013 [arch] = Generates Visual Studio 2013 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 11 2012 [arch] = Generates Visual Studio 2012 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 10 2010 [arch] = Deprecated.  Generates Visual Studio 2010
                                 project files.  Optional [arch] can be
                                 "Win64" or "IA64".
  Visual Studio 9 2008 [arch]  = Generates Visual Studio 2008 project files.
                                 Optional [arch] can be "Win64" or "IA64".
  Borland Makefiles            = Generates Borland makefiles.
  NMake Makefiles              = Generates NMake makefiles.
  NMake Makefiles JOM          = Generates JOM makefiles.
  MSYS Makefiles               = Generates MSYS makefiles.
  MinGW Makefiles              = Generates a make file for use with
                                 mingw32-make.
  Green Hills MULTI            = Generates Green Hills MULTI files
                                 (experimental, work-in-progress).
  Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - MinGW Makefiles = Generates CodeBlocks project files.
  CodeBlocks - NMake Makefiles = Generates CodeBlocks project files.
  CodeBlocks - NMake Makefiles JOM
                               = Generates CodeBlocks project files.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - MinGW Makefiles   = Generates CodeLite project files.
  CodeLite - NMake Makefiles   = Generates CodeLite project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Eclipse CDT4 - NMake Makefiles
                               = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - MinGW Makefiles
                               = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.
  Kate - MinGW Makefiles       = Generates Kate project files.
  Kate - NMake Makefiles       = Generates Kate project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Sublime Text 2 - MinGW Makefiles
                               = Generates Sublime Text 2 project files.
  Sublime Text 2 - NMake Makefiles
                               = Generates Sublime Text 2 project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.

Exception: C:\Users\cpyar\Desktop\obs\obs-plugintemplate\.github\scripts\Build-Windows.ps1:101
Line |
 101 |  Build
     |  ~~~~~
     | cmake -S . -B plugin_build_x64 -G  -DCMAKE_SYSTEM_VERSION=10.0.20348.0 -DCMAKE_GENERATOR_PLATFORM=x64
     | -DCMAKE_BUILD_TYPE=RelWithDebInfo -DQT_VERSION=6 -DENABLE_PLUGINS=OFF -DENABLE_UI=OFF
     | -DENABLE_SCRIPTING=OFF
     | -DCMAKE_INSTALL_PREFIX:PATH=C:\Users\cpyar\Desktop\obs\obs-build-dependencies\plugin-deps-2022-08-02-qt6-x64 -DCMAKE_PREFIX_PATH:PATH=C:\Users\cpyar\Desktop\obs\obs-build-dependencies\plugin-deps-2022-08-02-qt6-x64 exited with non-zero code 1.
```

### Steps to Reproduce

1. git clone obs-plugintemplate
2. cd into template directory
3.  execute  `.\.github\scripts\Build-Windows.ps1`
...

### Anything else we should know?

I just earlier cloned the obs-studio repo and successfully built obs using the build scripts. 
Then I downloaded the template and this is the result. other than that I cleaned up all previous instances of the build environment

**Comment by RytoEX** [comment]
You must specify a valid generator to `Build-Windows.ps1`. The valid generators are currently:
* Visual Studio 17 2022
* Visual Studio 16 2019

Use the `-CMakeGenerator` param of the build script to specify the generator name as a quoted string.

**Comment by PatTheMav** [comment]
By default the script should take the Visual Studio version as defined in `buildspec.json`, which uses "Visual Studio 17 2022" by default.

So this looks like a genuine bug to me.

**Comment by PatTheMav** [comment]
@cpyarger Should be fixed by https://github.com/obsproject/obs-plugintemplate/commit/50dd9b8973cd4faa91ed32d45cf1d0f2e1b27d2a, pls confirm.

**Comment by cpyarger** [comment]
> @cpyarger Should be fixed by [50dd9b8](https://github.com/obsproject/obs-plugintemplate/commit/50dd9b8973cd4faa91ed32d45cf1d0f2e1b27d2a), pls confirm.

Seems like it's fixed
